data-prepper-plugins/armeria-common/src/main/java/com/amazon/dataprepper/plugins/HttpBasicArmeriaHttpAuthenticationProvider.java

```diff
@@ -39,15 +39,14 @@ public HttpBasicArmeriaHttpAuthenticationProvider(final HttpBasicAuthenticationC
 
     @Override
     public void addAuthenticationDecorator(final ServerBuilder serverBuilder) {
-        serverBuilder.decorator(createDecorator());
+        serverBuilder.decorator("regex:^/(?!health$).*$", createDecorator());
```

Comment:
This change only applies to HttpBasicArmeriaHttpAuthenticationProvider here.

The solution has to be generic that we can apply to implementation of ArmeriaHttpAuthenticationProvider. This logic should not live in the AuthenticationProvider, it's responsibility is only to provide Authentication Implementation. This use case is specific to the source, for example HTTP source or OTel Source. The source should be able to configure which path to authenticate.

[ArmeriaHttpAuthenticationProvider](https://github.com/dapowers87/data-prepper/blob/49a16a8bae8002c2f2a898dc6b6edfde332e27f1/data-prepper-plugins/armeria-common/src/main/java/com/amazon/dataprepper/armeria/authentication/ArmeriaHttpAuthenticationProvider.java#L30-L31) needs to support returning a service decorator as supported in [GrpcAuthenticationProvider](https://github.com/opensearch-project/data-prepper/blob/main/data-prepper-plugins/armeria-common/src/main/java/com/amazon/dataprepper/armeria/authentication/GrpcAuthenticationProvider.java#L40)

We need to update the interface method
```
void addAuthenticationDecorator(ServerBuilder serverBuilder);
```

to

```
Function<? super HttpService, ? extends HttpService> getHttpAuthenticationService() 
```

In the current ArmeriaHttpAuthenticationProvider we are providing more control to the AuthProvider by passing in the ServerBuilder which is not required.


@dlvenable What is the implication of making this change ?